### PR TITLE
Retry TransportShardMultiGetFomTranslogAction during relocations

### PR DESCRIPTION
**Pull request by pxsalehi**
Same as https://github.com/elastic/elasticsearch/pull/104579 but for MGET.

Relates ES-5727

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java [hunk 1]
```diff
@@ -173,57 +182,115 @@ private void handleMultiGetOnUnpromotableShard(
                 refreshRequest,
                 listener.delegateFailureAndWrap((l, replicationResponse) -> super.asyncShardOperation(request, shardId, l))
             );
-        } else if (request.realtime()) {
```
Comment (pxsalehi) [inline server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java:176]: This branch has been moved as-is to `tryShardMultiGetFromTranslog`

File: server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java [hunk 2]
```diff
@@ -173,57 +182,115 @@ private void handleMultiGetOnUnpromotableShard(
...
+        ClusterStateObserver observer,
+        ActionListener<MultiGetShardResponse> listener
+    ) {
+        tryShardMultiGetFromTranslog(request, indexShard, state, listener.delegateResponse((l, e) -> {
+            final var cause = ExceptionsHelper.unwrapCause(e);
+            logger.debug("mget_from_translog[shard] failed", cause);
+            if (cause instanceof ShardNotFoundException
+                || cause instanceof IndexNotFoundException
+                || cause instanceof NoShardAvailableActionException
+                || cause instanceof UnavailableShardsException) {
```
Comment (ywangd) [inline server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java:216]: Based on this comment https://github.com/elastic/elasticsearch/pull/104579#discussion_r1472954697, I assume we should remove these two exceptions here as well and also make `getCurrentNodeOfPrimary` use listener?

**Review by ywangd — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java (hunk 2 above)
Comment (pxsalehi) [reply server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java:216]: ja!

**Pull request merged**